databricks/koalas/base.py

```diff
@@ -171,6 +173,11 @@ def spark_column(self):
     __neg__ = column_op(Column.__neg__)
 
     def __add__(self, other):
+        if not isinstance(self.spark.data_type, StringType):
+            if (
+                isinstance(other, IndexOpsMixin) and isinstance(other.spark.data_type, StringType)
+            ) or isinstance(other, str):
```

Comment:
Maybe I think we can do this also with single `if` statement like other methods ?

```python
        if not isinstance(self.spark.data_type, StringType) and (
            (isinstance(other, IndexOpsMixin) and isinstance(other.spark.data_type, StringType))
            or isinstance(other, str)
        ):
```

Reply:
Makes sense! Updated.